pkg/engine/engine.go

```diff
@@ -630,29 +628,6 @@ func (e *Engine) handleEdgeConstainConstraint(constraint *constraints.EdgeConstr
 	return nil
 }
 
-func (e *Engine) ApplyResourceConstraint(graph *construct.ResourceGraph, constraint *constraints.ResourceConstraint) EngineError {
```

Comment:
dont we either need this or line 368 to process resource constraints? Or am i missing that its moved somewhere else

Reply:
This function itself was no longer being used. The line 368 change just moved it into the function `configureResource` which processes the constraints. It also now occurs on resource adding via decisions to make sure that any new resources are configured properly according to both user input constraints (if they happen to add a constraint for something that doesn't _yet_ exist but would during resource expansion perhaps in the future we'd support `{{ downstream x }}` for example) as well as the templates' configuration.